File: epi_judge_python/int_as_array_increment.py
from typing import List

from test_framework import generic_test


def plus_one(arr: List[int]) -> List[int]:
    arr[len(arr) - 1] += 1
    for i in reversed(range(1, len(arr))):
        if arr[i] != 10:
            return arr
        arr[i] = 0
        arr[i - 1] += 1

    if arr[0] == 10:
        arr[0] = 1
        arr.append(0)

    return arr


if __name__ == '__main__':
    exit(
        generic_test.generic_test_main('int_as_array_increment.py',
                                       'int_as_array_increment.tsv', plus_one))

File: epi_judge_cpp/int_as_array_increment.cc
#include <vector>

#include "test_framework/generic_test.h"

using std::vector;

vector<int> PlusOne(vector<int> nums) {
    int carry = 1;
    for (int i = nums.size() - 1; i >= 0; i--) {
        int sum = nums[i] + carry;
        carry = sum / 10;
        nums[i] = sum % 10;
        if (carry == 0) break;
    }
    if (carry == 1) {
        nums.insert(nums.begin(), 1);
    }
    return nums;
}

int main(int argc, char *argv[]) {
    std::vector<std::string> args{argv + 1, argv + argc};
    std::vector<std::string> param_names{"A"};
    return GenericTestMain(args, "int_as_array_increment.cc",
                           "int_as_array_increment.tsv", &PlusOne,
                           DefaultComparator{}, param_names);
}
